app/Http/Controllers/RequestedSongController.php

```diff
@@ -68,9 +67,34 @@ public function upvote(Company $company, RequestedSong $requestedSong)
   /**
    * Store a newly created resource in storage.
    */
-  public function store(Request $request)
+  public function store(Company $company, Request $request)
   {
-    //
+    $requestedSong = $this->getTrackInfo($request->spotifyId);
+
+    $song = Song::updateOrCreate([
+      'spotify_id' => $requestedSong['spotify_id']
+    ], $requestedSong);
```

Comment:
```suggestion
    $song = Song::where(['spotify_id' => $requestedSong['spotify_id']])
        ->firstOr(function() use ($request) {
            return Song::create($this->getTrackInfo($request->spotifyId));
        });
```

![image](https://github.com/purplena/playlist-app/assets/7047375/211cf0d9-3e95-4ea9-81a9-84937d258624)